### PR TITLE
[v1.13] Bump envoy to v1.27.x

### PR DESCRIPTION
**Pull request by sayboras**
* [x] #26101 (@sayboras)
* [x] #28126  (@jrajahalme).
* [x] #27498 (@jrajahalme )
* [x] #31221 (@sayboras)

### CODE REVIEW

**Comment by sayboras** [comment]
/test-backport-1.13

**Comment by sayboras** [comment]
/test-1.16-4.19

**Comment by sayboras** [comment]
/test-1.21-4.19

**Comment by sayboras** [comment]
Remove preview only label as https://github.com/cilium/cilium/releases/tag/v1.13.14 is released.

**Comment by sayboras** [comment]
/test-backport-1.13

**Comment by sayboras** [comment]
/test-1.21-4.19

**Comment by sayboras** [comment]
/test-1.24-4.19

File: pkg/policy/api/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ package api
 import (
 	labels "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels"
 	v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
-	kafka "github.com/cilium/cilium/pkg/policy/api/kafka"
+	kafka "github.com/cilium/proxy/pkg/policy/api/kafka"
```
Comment (learnitall) [inline pkg/policy/api/zz_generated.deepcopy.go:14]: Was this line manually changed? Or did you use the Makefile to change this?

**Review by learnitall — COMMENTED** [review]
Quick clarifying question, otherwise looks good for API changes. If you need a more advanced review from an API perspective, I'd recommend looping other folks in- my knowledge of our proxy stack isn't good enough to give a lot of feedback here.

File: pkg/policy/api/zz_generated.deepcopy.go (hunk 1 above)
Comment (sayboras) [reply pkg/policy/api/zz_generated.deepcopy.go:14]: it was changed manually, find and replace in IDE helps a lot though.

**Review by learnitall — APPROVED** [review]

**Comment by jrajahalme** [comment]
@sayboras Optimally we would not introduce functional changes to v1.13. Did you find that we had introduced breaking changes to Envoy so that this update did not work our without the Ingress IP?

Seems that the drop privileges change is wanted to avoid reverting that change on `cilium-envoy` side, and there has been sufficient testing on it so that we are fairly confident that it will not cause any issues.

**Review by jrajahalme — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

**Comment by julianwiedmann** [comment]
@sayboras please manually update the `backport-done` label for the affected PRs :)

**Pull request merged**